client/components/Room.jsx

```diff
@@ -1,29 +1,37 @@
 import React from 'react';
 
 import Writer from "../components/Writer.jsx"
+import Loader from "../components/Loader.jsx"
 
-const userIdToProfile = function(chatterUsers) {
-  const idToProfile = {};
-  chatterUsers.forEach( (user) => {
-    idToProfile[user._id] = {
-      nickname: user.nickname,
-      avatar: user.avatar
-    };
-  });
-  return idToProfile;
+
+const isFirstMessage = function(prevMsg, currentMsg) {
+  return prevMsg.userId != currentMsg.userId;
+};
+
+const timeSinceLastMsgGreaterThan = function(minutes, prevMsg, currentMsg) {
+  const difference = currentMsg.createdAt - prevMsg.createdAt;
+  const resultInMinutes = Math.round(difference / 60000);
+  return resultInMinutes > minutes;
 };
 
+const timestampShouldBeDisplayed = function(currentMsg, nextMsg) {
+  const veryRecentMessage = currentMsg.getMinutesAgo() <= 60;
+  const recentMessage = currentMsg.getMinutesAgo() <= 1440;
+  return veryRecentMessage && timeSinceLastMsgGreaterThan(2, currentMsg, nextMsg) || recentMessage && timeSinceLastMsgGreaterThan(60, currentMsg, nextMsg);
+}
+
+const roomSubs = new SubsManager();
+
 const Room = React.createClass({
   mixins: [ReactMeteorData],
 
   getMeteorData () {
     const { roomId } = this.props;
-    const messagesHandle = Meteor.subscribe("chatterMessages", {
-      roomId: roomId,
-      messageLimit: Chatter.options.messageLimit
+    const messagesHandle = roomSubs.subscribe("chatterMessages", {
+      roomId: roomId
```

Comment:
I think I have learned this haha. Will keep eyes open if I see any unchanged ones.